### DISCUSSION TITLE
Transfer Same Bank vs. Transfer Specific Banks

### DISCUSSION THREAD

**Discussion by cparke2** [bisq-network/bisq#7256] [General]
Why does Bisq have both of these payment method types?  They're really confusing, and somewhat interchangeable, so it's really strange.

1. "Transfer Same Bank" is rather straightforward: make and take offers require the same bank (intra-bank transfers)
2. "Transfer Specific Banks" is a hybrid. When you make an offer, you provide a list of banks that you will accept payment at. However, when taking an offer, you still must designate on the payment method just one bank that you will be using to make payment (which could be a totally different bank). 
3. For a "Transfer Specific Banks" offers to match your list of accounts, you must have a "Transfer Specific Banks" account established; it won't match with any "Transfer Same Bank" even if the same bank name is designated.
4. While "Transfer Specific Banks" permits several banks to be listed in a single offer, the list of banks is limited to like 100 characters, which is not enough for some intra-bank networks like the Real-Time Payments (RTP) Network in U.S.A. which have more banks than would fit in the allowed space, but you're also not allowed to clone your "specific banks" offer with a different "Transfer Specific Banks" account that lists entirely different banks. This means payments through an unsupported network are limited unless/until Bisq adds it as a separate payment method?

A few possible resolutions:
1. "Transfer Same Bank" accounts should match "Transfer Specific Banks" for taking offers when the "same bank" is included in the "specific banks" list. However, this would be an account-type mismatch on a trade.
2. "Transfer Specific Banks" should not be allowed to take offers at all, or it should allow taking any "same bank" or "specific bank" offer with a bank name in common.

The status quo may work if you set up many duplicate "Transfer Specific Banks" accounts with a each specific banks as a different "taker payment" account, and "Transfer Same Bank" for each bank, but I really see these two payment methods as related and not sure why all this functionality isn't combined into a single "Transfer Same or Specific Banks" payment method.

**Comment by HenrikJannsen** [comment]
"Transfer Same Bank" is useful in countries when there are fees between banks but not if both use the same bank. 
"Transfer Specific Banks" is an extension to a group of banks. I also don't see a reason why "Transfer Same Bank" could not be just a special case of "Transfer Specific Banks" with only one entry. But not 100% sure if there are other reasons to keep them separated.
Changes though would need to be backward compatible which makes that all quite tricky and comes with significant effort.

Most Bisq devs are working now on Bisq2 and the payment methods design there can be built in a better and more flexible manner. If you are interested in that join the matrix channels and PM me there.
